include/openthread/platform/radio.h

```diff
@@ -1366,6 +1366,18 @@ extern otError otPlatRadioGetRawPowerSetting(otInstance *aInstance,
                                              uint8_t    *aRawPowerSetting,
                                              uint16_t   *aRawPowerSettingLength);
 
+/**
+ * Set the IEEE EUI-64 for this transceiver, if the OTP is not supported.
+ *
+ * @param[in]  aInstance   The OpenThread instance structure.
+ * @param[in]  aIeeeEui64  The IEEE EUI-64.
+ *
+ * @retval  OT_ERROR_NONE           Successfully.
+ * @retval  OT_ERROR_INVALID_STATE  Failed if the EUI64 is set by the factory.
+ *
+ */
+otError otPlatRadioSetIeeeEui64(otInstance *aInstance, const otExtAddress *aIeeeEui64);
```

Comment:
I am not in favor of adding  this API in `otPlatRadio`. 

If a plaform/radio need to set this, it may be realized using a vendor specific API (outside of OT stack API) with a spinel vendor extension and vendor specific  spinel property.

Reply:
In which folder / file i can do this specific vendor API (outside of OT stack API) with a spinel vendor extension and vendor specific spinel property ?

Reply:
Note that this will be then vendor specific code and not part of OpenThread repo. 

- Spinel NCP code allows vendors to define vendor-specific commands and properties. You can check [`ncp/example_vendor_hook.cpp`](https://github.com/openthread/openthread/blob/12af0231215d34b751c8dc318fa11d77dbfefb9e/src/ncp/example_vendor_hook.cpp)
- The `RadioSpinel` also provides vendor hooks [`spinel/example_vendor_hook.hpp`](https://github.com/openthread/openthread/blob/12af0231215d34b751c8dc318fa11d77dbfefb9e/src/lib/spinel/example_vendor_hook.hpp)

The reason I am not in favor of adding this as a `otPlatRadio` API is that it can confuse platform developers. The EUI64 is intended to be fixed and factory-assigned and having a general API in OT stack that allow it to be set is not a good idea.

Reply:
Agree with @abtink, I think you also can add a vendor diag command `diag eui64 xxx` to the platform layer to set the EUI64. Similar to https://github.com/openthread/ot-nrf528xx/blob/main/src/src/diag.c#L324.